pkg/crd/gen.go

```diff
@@ -134,30 +134,58 @@ func (g Generator) Generate(ctx *genall.GenerationContext) error {
 		GenerateEmbeddedObjectMeta: g.GenerateEmbeddedObjectMeta != nil && *g.GenerateEmbeddedObjectMeta,
 	}
 
+	if ctx.Logger != nil {
```

Comment:
Is there such an implementation as a null logger? I wonder if it would be better to extract the logger at the start of the function, and init it if nil to a null logger.

Then we don't have to do the nil check at every call site?